src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -43,7 +51,9 @@ public class EditCommand extends Command {
             + "[" + PREFIX_PHONE + "PHONE] "
             + "[" + PREFIX_EMAIL + "EMAIL] "
             + "[" + PREFIX_ADDRESS + "ADDRESS] "
-            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "[" + PREFIX_TAG + "TAG]"
+            + "[" + PREFIX_HOURS + "/" + PREFIX_DONATED_AMOUNT + "/" + PREFIX_PARTNERSHIP_END_DATE
+            + "HOURS/AMOUNT/DATE]...\n"
```

Comment:
Is there an error that would be thrown if the user was a `Volunteer` and I tried to edit the partnership end date instead? Or what would be the behaviour

Reply:
Currently no errors will be thrown. But it does not break the code either. Only relevant info field will be displayed( i.e. Hours for volunteers, even if somehow it has a DonatedAmount value. Will be refined in later iterations.

Reply:
Would it be the intended behaviour then, for no errors to be displayed ,even if I tried editing an attribute of a Person that does not have the correct role?

Since each push should result in a working product, I am not entirely convinced this would be an intended behaviour of our working product? 

Do chime in with your thoughts @Justin-Yeo @voldiemort @chongsun2002

Reply:
Would it be better to be more flexible and allow refinements to be made in future commits? If everything has to be perfect in each push, it would not be easy to reduce code amount which makes code reviewing easier as you mentioned.

Reply:
Technically, it is a working product. I think we should distinguish between app breaking down and imperfect behaviors.